utils_cv/detection/dataset.py

```diff
@@ -96,38 +150,64 @@ def __init__(
             self.train_ds,
             batch_size=self.batch_size,
             shuffle=True,
-            num_workers=4,
+            num_workers=db_num_workers(),
             collate_fn=collate_fn,
         )
 
         self.test_dl = DataLoader(
             self.test_ds,
             batch_size=self.batch_size,
             shuffle=False,
-            num_workers=4,
+            num_workers=db_num_workers(),
             collate_fn=collate_fn,
         )
 
-    def _get_labels(self) -> List[str]:
+    def _read_annos(self) -> List[str]:
         """ Parses all Pascal VOC formatted annotation files to extract all
         possible labels. """
-        # TODO implement:
-        # - option to bypass this setup set if users already know label names
-        # - option to do 'active' loading so that we preload all annotations
-        # during setup time (instead of lazy loading) - this will let us
-        # do things like get average box size of all the data, etc..
 
+        # If im_dir specified then get image paths and respective annotations
+        # from scanning that directory. Otherwise find all annotation files
+        # and get path to image from annotation.
+        if self.im_dir is None:
```

Comment:
currently, you have to know that this code prioritizes using im_dir over a path in the annotation file. If you're just using this function, you won't know this. 

I would propose raising an error if BOTH im_dir is specified and `parse_pascal_voc_anno` returns an `im_path`